Automattic/jetpack
# WAF: Add Waf_Stats class

#### Description
Adds a `Waf_Stats` class to the WAF package for generating an array of WAF stats.

#### Changes proposed in this Pull Request:

* Adds a `Waf_Stats` class to the WAF package
* Adds methods for retrieving each of the following:
   * IP allow list count
   * IP block list count
   * Rules version
   * Date the automatic rules were last updated
* Adds these values to the `waf.stats` property in the Protect initial state

#### Other information:

#### Jetpack product discussion


- 1202232924268038-as-1203282049665716

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:





* Checkout this branch
* Add a Scan upgrade
* Activate Protect, and proceed to the Firewall tab
* console.log( window.jetpackProtectInitialState )
* Verify that the output contains the `state.waf.stats` property and that value is an array of the expected stats
* Manually disable the WAF module and log the initial state again
* Verify that the `stats` property displays a default value
* Validate that there is no regression in app functionality introduced
